functions.php

```diff
@@ -839,40 +839,93 @@ function display_rich_snippet( $content ) {
 		return ( is_single() || is_page() ) ? $content . $software : $content;
 	} elseif ( '9' == $type ) {
 		global $post;
+
+		if ( ! isset( $post->ID ) ) {
+			return;
+		}
+
 		$args_video = get_option( 'bsf_video' );
 		$video      = '';
 
+		// Generate the schema structure.
+		$video_title    = get_post_meta( $post->ID, '_bsf_video_title', true );
+		$video_desc     = get_post_meta( $post->ID, '_bsf_video_desc', true );
+		$video_thumb    = get_post_meta( $post->ID, '_bsf_video_thumb', true );
+		$video_url      = get_post_meta( $post->ID, '_bsf_video_url', true );
+		$video_emb_url  = get_post_meta( $post->ID, '_bsf_video_emb_url', true );
+		$video_duration = get_post_meta( $post->ID, '_bsf_video_duration', true );
+		$video_date     = get_post_meta( $post->ID, '_bsf_video_date', true );
+
+		// Set default timezone and handle video date.
+		if ( '' != trim( $video_date ) ) {
+			// Get the server timezone.
+			$timezone = date_default_timezone_get();
+
+			// Check if $video_date is set and valid, and $timezone is set.
+			if ( ! empty( $video_date ) && isset( $timezone ) ) {
+				try {
+					// Create a DateTime object from the $video_date string.
+					$datetime   = new DateTime( $video_date, new DateTimeZone( $timezone ) ); // Set the timezone to the server's timezone.
+					$uploadDate = $datetime->format( 'Y-m-d\TH:i:sP' ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+				} catch ( Exception $e ) {
+					// Translators: %s is the error message from the exception.
+					echo esc_html( sprintf( __( 'Error creating DateTime object: %s', 'text-domain' ), esc_html( $e->getMessage() ) ) );
```

Comment:
'text-domain' should be actual domain of our plugin,

never copy paste anything blindly

Reply:
sorry was checking something in hurry , will be mindful moving forward